ttcn3/types/types.go

```diff
@@ -399,5 +420,32 @@ func printExpr(e syntax.Expr) string {
 }
 
 func TypeOf(n syntax.Expr) Type {
+	//types to check:
+	//ValueLiteral, CompositeLiteral, BinaryExpr, UnaryExpr, Ident
+	switch n := n.(type) {
+	case *syntax.ValueLiteral:
+		kind := kindConvert[n.Tok.Kind()]
+		switch kind {
+		case Integer, Float, Boolean:
+			return Predefined[kind.String()]
+		case Verdict:
+			return Predefined["verdicttype"]
```

Comment:
Without conversion table the code almost looks the same:
```suggestion
		switch n.Tok.Kind() {
		case syntax.INT:
			return Predefined["integer"]
		case syntax.FLOAT:
		        return Predefined["float"]
		 case syntax.TRUE, syntax.FALSE:
		        return Predefined["boolean"]
```

I would go with that.